### ISSUE TITLE
Tags/Categories Hierarchy Parsing

### ISSUE THREAD

**Issue by Zaroz** [janeczku/calibre-web#2019]
Hello, I would like to request that the app be able to parse Hierarchical tags. What I mean by Hierarchical tags is the function in calibre where you can nest tags under other tags by using a period (.) separator (i,e: Fantasy.Magic). 

As it is right now, a calibre tag of "Fantasy.Magic" shows up in the Calibre-Web Category browser under the Letter "F" for Fantasy rather than "M" for Magic. 

![image_2021-06-09_202457](https://user-images.githubusercontent.com/22653076/121450131-c1c6b280-c960-11eb-8f10-2f07652e706f.png)

This makes it rather difficult to search for some tags as there is a lot of them that fall under the "Fantasy" parent tag and some are even nested under several more levels, and you would need to know the whole hierarchy to properly find it. 

Additionally, when listing tags in the book details page, it displays the full tag path for each tag, taking up a lot of space:
![image_2021-06-09_202227](https://user-images.githubusercontent.com/22653076/121449946-685e8380-c960-11eb-933a-8f53a0f683d9.png)

What I would like to see is that the app would separately display tags for each level of the hierarchy under their respective letters in the Categories browser and collapse the displayed tags in the book details page to the bottom 1 or 2 levels of the hierarchy in a cleaner format, with maybe hover text displaying the full path (i.e: for Fantasy.Otherworld.Isekai, just display "Otherworld > Isekai" or just "Isekai" on the book details page.

**Comment by tinywrkb** [comment]
Dup of #1956 and ##1185 is also relevant. This has a better description of the wanted feature so keep this and close at least #1956.

I would also like to see this being implemented.

**Comment by Zaroz** [comment]
> Dup of #1956 and ##1185 is also relevant. This has a better description of the wanted feature so keep this and close at least #1956.
> 
> I would also like to see this being implemented.

Not really, a dup of #1956, but related I guess. I feel like he was more concerned about searching using AND logic rather than OR logic, which you can already do using advanced search.

**Comment by Subbeh** [comment]
Replying to get some traction, hopefully.

I second this request. Would be great to be able to use custom columns defined in Calibre, so we can have hierarchical categories.

**Comment by grisha-raykov** [comment]
Would love to have this feature.

**Comment by machmalpause** [comment]
I also use "." separated attributes for grouping of my tons of PDFs into hierarchical categories. I think this is a MUST for huge libraries.
It would be fantastic if a proper visualisation of this feature could be soon incorporated in calibre web (which I love for its smooth performance on mobile devices) !

**Comment by stathisstef** [comment]
I would love to see this implemented. Nested tags really feels like the bow on top of a large and well-organized calibre library.

**Comment by retrodans** [comment]
I have just hit this today, whilst I have a nicely organised Calibre library, and it would be great if the hierarchical tags were presented to me in the UI in some way via calibre-web
